hil/ext/switches/juniper.py

```diff
@@ -0,0 +1,373 @@
+        set_native_vlan = self.set_native_vlan
+        if port_info[port]['trunk_mode']:
+            pass
+        else:
+            set_native_vlan = self.set_trunk_mode + set_native_vlan
+
+        if port_info[port]['disabled']:
+            set_native_vlan = self.enable_interface + set_native_vlan
+
+        if 'default' in port_info[port]['vlans']:
+            set_native_vlan = set_native_vlan + self.remove_default_vlan
+
+        self._set_commit_config(set_native_vlan)
+        self._clear_set_commands()
+
+    def _remove_native_vlan(self, port):
+        """Removes native vlan from a trunked port.
+        If it is the last vlan to be removed, it disables the port and
+        reverts its state to default configuration
+        """
+        port_info = self._interface_info(port)
+        vlan_id = str(port_info[port]['native_vlan'])
+        self._set_commands(port, vlan_id)
+        remove_native_vlan = self.remove_native_vlan
+        if isinstance(port_info[port]['vlans'], (str, unicode)):
+            self.revert_port(port)
+        else:
+            self._set_commit_config(remove_native_vlan)
+        self._clear_set_commands()
+
+    def _get_mode(self, port):
+        """Returns True if the port is in trunk mode else returns False. """
+        port_info = self._interface_info(port)
+        return port_info[port]['trunk_mode']
+
+    def _interface_info(self, port):
+        """Fetches latest committed configuration information about `port`"""
+        jun = self._create_session()
+        try:
+            with jun:
+                interface_config = InterfaceConfigTable(jun)
+                interface_config.get(
+                        interface=port, options={'database': 'committed'}
+                        )
+                ic_dict = json.loads(interface_config.to_json())
+        except ConnectError as err:
+                print("cannot connect to device: {0}".format(err))
+                sys.exit(1)
+
+        return ic_dict
+
+    def _get_port_networks(self, port):
+        """ Returns list of all the vlans shared with the port. """
+        port_info = self._interface_info(port)
+        return port_info[port]['vlans']
+
+    def _add_vlan_to_trunk(self, port, vlan_id):
+        """ Adds vlans to a trunk port. """
+
+        self._set_commands(port, vlan_id)
+        port_info = self._interface_info(port)
+        add_vlan = self.add_vlan
+
+        if not port_info[port]['trunk_mode']:
+            add_vlan = self.set_trunk_mode + add_vlan
+
+        if port_info[port]['disabled']:
+            add_vlan = self.enable_interface + add_vlan
+
+        if 'default' in port_info[port]['vlans']:
+            add_vlan = add_vlan + self.remove_default_vlan
+        self._set_commit_config(add_vlan)
+        self._clear_set_commands()
+
+    def _remove_vlan_from_port(self, port, vlan_id):
+        """ removes a single vlan specified by `vlan_id` """
+#        import pdb; pdb.set_trace()
```

Comment:
should be removed

Reply:
it is not in my file. I guess some update issue. I will commit again.